lib/deimos/handler.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+require 'deimos/consume/batch_consumption'
+require 'deimos/consume/message_consumption'
+
+# Class to consume messages coming from a Kafka topic
+# Note: According to the docs, instances of your handler will be created
+# for every incoming message/batch. This class should be lightweight.
+module Deimos
+  # Basic consumer class. Inherit from this class and override either consume
+  # or consume_batch, depending on the delivery mode of your listener.
+  # `consume` -> use `delivery :message` or `delivery :batch`
+  # `consume_batch` -> use `delivery :inline_batch`
+  class Handler
+    include Consume::MessageConsumption
+    include Consume::BatchConsumption
```

Comment:
I see this class as being much more lightweight - it should be a simple Phobos listener which calls the consumer class with the messages it received. The tracing etc. should still happen on the consumer side because we want the metrics to be tagged by topic.